org.eclipse.jdt.core.tests.model/src/org/eclipse/jdt/core/tests/model/JavaSearchBugsTests.java

```diff
@@ -13182,6 +13182,44 @@ public void testBug355605() throws CoreException {
 		deleteProject("P");
 	}
 }
+public void testBug355605_2() throws CoreException {
+	try {
+		createJavaProject("P");
+
+		String fileContent =
+				"public class X { \n"
+						+ "class R {\n"
+						+ "   class S {\n"
+						+ "     S(String s) {}\n"
+						+ "   	void setInfo(String x) {\n"
+						+ "   	}\n"
+						+ "   }\n"
+						+ "   class T {\n"
+						+ "   }\n"
+						+ "	T t = new T()  {\n"
+						+ "		S s = new S(\"test\") {\n"
```

Comment:
As commented on the PR, the string we are changing must match `X.R.1.new T(){}.1.new S(){}` during this test.

I've added a paranoia test (copy of the one above) with an argument in the constructor of `S`. Just to be sure the argument doesn't make it into the string `X.R.1.new T(){}.1.new S(){}`. It doesn't, but best to be sure with a test.